Runtime/Triangulator.cs

```diff
@@ -173,187 +234,217 @@ public class InputData
+            static readonly ProfilerMarker MarkerValidateInput = new ProfilerMarker("ValidateInput");
+            static readonly ProfilerMarker MarkerDelaunayTriangulation = new ProfilerMarker("DelaunayTriangulation");
+            static readonly ProfilerMarker MarkerConstrainEdges = new ProfilerMarker("ConstrainEdges");
+            static readonly ProfilerMarker MarkerPlantSeeds = new ProfilerMarker("PlantSeeds");
+            static readonly ProfilerMarker MarkerRefineMesh = new ProfilerMarker("RefineMesh");
+            static readonly ProfilerMarker MarkerInverseTransformation = new ProfilerMarker("InverseTransformation");
+
+            static void PreProcessInput (Preprocessor preprocessor, InputData input, OutputData output, out NativeList<float2> localPositions, out NativeArray<float2> localHoles, out AffineTransform2D localTransformation) {
+                localPositions = output.Positions.IsCreated ? output.Positions : new NativeList<float2>(input.Positions.Length, Allocator.Temp);
+                localPositions.ResizeUninitialized(input.Positions.Length);
+                if (preprocessor == Preprocessor.PCA || preprocessor == Preprocessor.COM) {
+                    localTransformation = preprocessor == Preprocessor.PCA ? CalculatePCATransformation(input.Positions) : CalculateLocalTransformation(input.Positions);
+                    localTransformation.Transform(input.Positions, localPositions.AsArray());
+                    if (input.HoleSeeds.IsCreated) {
+                        localHoles = new NativeArray<float2>(input.HoleSeeds.Length, Allocator.Temp);
+                        localTransformation.Transform(input.HoleSeeds, localHoles);
+                    } else {
+                        localHoles = default;
+                    }
+                } else if (preprocessor == Preprocessor.None) {
+                    localPositions.CopyFrom(input.Positions);
+                    localHoles = input.HoleSeeds;
+                    localTransformation = AffineTransform2D.identity;
+                } else {
+                    throw new System.ArgumentException();
+                }
+            }
+
+            public void Execute() {
+                output.Status.Value = Status.OK;
+                output.Triangles.Clear();
+                output.Positions.Clear();
+
+                MarkerPreProcess.Begin();
+                PreProcessInput(preprocessor, input, output, out var localPositions, out var localHoles, out var localTransformation);
+                MarkerPreProcess.End();
+
+                if (validateInput) {
+                    MarkerValidateInput.Begin();
+                    ValidateInput(localPositions.AsArray(), input.ConstraintEdges);
+                    MarkerValidateInput.End();
+                }
+
+                MarkerDelaunayTriangulation.Begin();
+                var halfEdges = new NativeList<int>(localPositions.Length, Allocator.Temp);
```

Comment:
I would suggest to stick with the notation. I mean to name it `halfedges` instead of `halfEdges`. Please check the rest of the code too.